app/helpers/schemes_helper.rb

```diff
@@ -60,10 +60,10 @@ def selected_schemes_and_locations_text(download_type, schemes)
     when "schemes"
       "You've selected #{pluralize(scheme_count, 'scheme')}."
     when "locations"
-      location_count = schemes.map(&:locations).flatten.count
+      location_count = Location.where(scheme: schemes).count
```

Comment:
Previously was very inefficient for large numbers of schemes, so the "download csv" was taking ages to load